tf2_ros/include/tf2_ros/buffer_interface.h

```diff
@@ -66,7 +66,7 @@ namespace tf2_ros
   {
     auto ns = std::chrono::duration<double, std::nano>(time_msg.nanosec);
     auto s = std::chrono::duration<double>(time_msg.sec);
-    return (s + std::chrono::duration_cast<std::chrono::seconds>(ns)).count();
+    return (s.count() + 1e-9 * std::chrono::duration_cast<std::chrono::nanoseconds>(ns).count());
```

Comment:
Thanks for investigation and fixing! To avoid the constant `1e-9` in code, may I suggest this instead?

```
return (s.count() + std::chrono::duration_cast<std::chrono::duration<double>>(ns).count());
```

Reply:
It's better, thank you for review.